### PR TITLE
deprecation & other minor fixes for Xcode 13.4

### PR DESCRIPTION
**Pull request by getaaron**
- add `.ds_store` to gitignore
- cocoapods was configured only for swiftlint; i removed that and added swiftlint via mint
- fixed a few xcode warnings (class -> anyobject)
- moved headers build phase up to avoid intermittent build cyclical dependency problem
- removed `swiftlint:disable:this unused_setter_value` as swiftlint no longer flags this

### CODE REVIEW

File: Example-iOS/ViewControllers/DynamicTableViewController.swift [hunk 1]
```diff
@@ -45,7 +45,7 @@ internal final class DynamicViewController: QuickTableViewController {
     get {
       return quickTableView
     }
-    set {} // swiftlint:disable:this unused_setter_value
+    set {}
```
Comment (getaaron) [inline Example-iOS/ViewControllers/DynamicTableViewController.swift:48]: this was fixed in SwiftLint 0.46 (CI is currently running 0.40 and this PR updates it to 0.48)

https://github.com/realm/SwiftLint/issues/2585

File: Example-iOS/ViewControllers/DynamicTableViewController.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ internal final class DynamicViewController: QuickTableViewController {
     get {
       return cachedTableContents
     }
-    set {} // swiftlint:disable:this unused_setter_value
+    set {}
```
Comment (getaaron) [inline Example-iOS/ViewControllers/DynamicTableViewController.swift:39]: this was fixed in SwiftLint 0.46 (CI is currently running 0.40 and this PR updates it to 0.48)

https://github.com/realm/SwiftLint/issues/2585

File: Example-iOS/ViewControllers/DynamicTableViewController.swift (hunk 1 above)
Comment (getaaron) [inline Example-iOS/ViewControllers/DynamicTableViewController.swift:48]: this was fixed in SwiftLint 0.46 (CI is currently running 0.40 and this PR updates it to 0.48)

https://github.com/realm/SwiftLint/issues/2585

File: Example-iOS/ViewControllers/DynamicTableViewController.swift (hunk 2 above)
Comment (getaaron) [inline Example-iOS/ViewControllers/DynamicTableViewController.swift:39]: this was fixed in SwiftLint 0.46 (CI is currently running 0.40 and this PR updates it to 0.48)

https://github.com/realm/SwiftLint/issues/2585

**Review by bcylin — APPROVED** [review]
@getaaron thanks much for updating this! Changes look good 👍 

The `unused_setter_value` warnings were generated by [houndci](https://github.com/apps/hound). It's still using SwiftLint `0.43.1` according to http://help.houndci.com/en/articles/2461415-supported-linters

I can't seem to find a way to configure houndci with SwiftLint version so I just added `unused_setter_value` to the disabled_rules.

The [build script](https://github.com/bcylin/QuickTableViewController/blob/da39e7754b0126f48eba90feca8baaf3a8df86bb/scripts/swiftlint.rb#L8) also needs some tweaks after moving away from CocoaPods. I'll update it later.

**Pull request merged**